### PR TITLE
Step 5: Internal logger

### PR DESCRIPTION
**Pull request by divbyzero**
- Added setter logLevel to EvolvConfig
- Moved Extension.swift to Example
- Added TestData file to Tests
- Added test for Execution
- Added test for logger

### CODE REVIEW

File: Source/DefaultEvolvHttpClient.swift [hunk 1]
```diff
@@ -47,14 +47,13 @@ public class DefaultEvolvHttpClient: EvolvHttpClient {
             parameters: nil,
             encoding: JSONEncoding.default ,
             headers: headers).responseData { dataResponse in
-                self.logger.log(.debug, message: String(describing: dataResponse.request))
-                self.logger.log(.debug, message: String(describing: dataResponse.response))
+                self.logger.debug(dataResponse.request)
```
Comment (phylliswong) [inline Source/DefaultEvolvHttpClient.swift:50]: lines 50, 51 should be logging to the console, but sometimes does not. 
this is different from the previous log which consistently logged each time the checkout button was clicked. The flow is: client.emitEvent => eventEmitter.emit => makeEventRequest => httpClient.sendEvents => logger.debug

File: Source/EvolvLogger.swift [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        }
+
+        logMessage = String(format: "%@%@%@", basePrefix, logLevel.prefix(), "\(item)")
+        
+        DispatchQueue.global(qos: .utility).async { [weak self] in
+            guard let logMessage = self?.logMessage else {
+                return
+            }
+            
+            NSLog(logMessage)
```
Comment (phylliswong) [inline Source/EvolvLogger.swift:68]: got a bad instruction error here. steps to reproduce:
1. download app from this branch
2. pod install
3. build 
4. run
5. on the 3rd time clicking the checkout button, app crashes at this point

File: Source/EvolvLogger.swift [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+//  EvolvKit_Example
+//
+//  Created by phyllis.wong on 7/3/19.
+//  Copyright © 2019 CocoaPods. All rights reserved.
+//
+
+import Foundation
+
+public enum EvolvLogLevel: Int {
+    case info
```
Comment (phylliswong) [inline Source/EvolvLogger.swift:12]: thanks for updating this

File: Tests/ExecutionTest.swift [hunk 4]
```diff
@@ -12,23 +12,32 @@ import XCTest
 class ExecutionTest: XCTestCase {
```
Comment (phylliswong) [inline Tests/ExecutionTest.swift:12]: This is really great, thanks for creating this test. 
Non-blocking comment: we will need to support json values, even though that is not in this implementation. Can we add that in with the SwiftyJSON refactor please?

File: Tests/ExecutionTest.swift [hunk 5]
```diff
@@ -122,5 +143,54 @@ class ExecutionTest: XCTestCase {
...
+        let defaultDoubleValue: Double = 1
+        let defaultFloatValue: Float = 1
+        let defaultBoolValue = true
+        
+        // when
+        let executionString = EvolvExecution(key: key, defaultValue: defaultStringValue, participant: mockParticipant, closure: closureString)
+        let executionInt = EvolvExecution(key: key, defaultValue: defaultIntValue, participant: mockParticipant, closure: closureInt)
+        let executionDouble = EvolvExecution(key: key, defaultValue: defaultDoubleValue, participant: mockParticipant, closure: closureDouble)
+        let executionFloat = EvolvExecution(key: key, defaultValue: defaultFloatValue, participant: mockParticipant, closure: closureFloat)
+        let executionBool = EvolvExecution(key: key, defaultValue: defaultBoolValue, participant: mockParticipant, closure: closureBool)
```
Comment (phylliswong) [inline Tests/ExecutionTest.swift:161]: Non-blocking: see comment above. Will need to support JSON as a data type.

File: Source/EvolvExecutionQueue.swift [hunk 6]
```diff
@@ -10,20 +10,21 @@ import SwiftyJSON
...
     static let shared = EvolvExecutionQueue()
     
-    public func enqueue<T>(_ execution: EvolvExecution<T>) {
+    func enqueue<T>(_ execution: EvolvExecution<T>) {
         queue.insert(execution, at: 0)
         count += 1
     }
     
-    public func executeAllWithValues(from rawAllocations: EvolvRawAllocations) throws {
+    func executeAllWithValues(from rawAllocations: EvolvRawAllocations) throws {
```
Comment (phylliswong) [inline Source/EvolvExecutionQueue.swift:27]: Non-blocking: current implementation does not support json as a data type, but needs to. 
Can we please add this  functionality with the refactor of SwiftyJSON?

**Comment by phylliswong** [comment]
Everything looks good. I added some comments about a small bug in the new logger, but can be resolved in the next PR to keep things moving.

File: Tests/ExecutionTest.swift (hunk 4 above)
Comment (divbyzero) [reply Tests/ExecutionTest.swift:12]: Yep, I'll add it.

File: Source/DefaultEvolvHttpClient.swift (hunk 1 above)
Comment (divbyzero) [reply Source/DefaultEvolvHttpClient.swift:50]: I found a problem with 
[EvolvKit][Error] Error sending data to Evolv (url: https://participants.evolv.ai/v1/sandbox/events?uid=sandbox_user&sid=BA3DA088-DE84-4E09-A735-77276CF30FF3&eid=experiment_1&cid=candidate_1&type=confirmation, code: 400, error: SUCCESS: {
    msg = "\"confirmation\" and \"contamination\" events must contain a valid cid.";
})

Could you suggest how to make valid cid?

**Pull request merged**